database/interface.go

```diff
@@ -389,6 +389,38 @@ type Tx interface {
 	// implementations.
 	FetchBlockRegions(regions []BlockRegion) ([][]byte, error)
 
+	// GetBlockCount counts all blocks stored in the database.
+	//
+	// Depending on the backend implementation, this function may have to
+	// iterate a cursor over all blocks and thus be relatively slow.
+	//
+	// The interface contract guarantees at least the following errors will
+	// be returned (other implementation-specific errors are possible):
+	//   - ErrTxClosed if the transaction has already been closed
+	GetBlockCount() (uint, error)
```

Comment:
I sort of think uint32 is more appropriate. We encode block heights as uint32 or int32 everywhere and with checkpoints, the number of blocks should never really come close to 4 billion. Also, if it did we'd have big problems because we use this number to allocate a slice of block nodes.

Reply:
Agreed.  I was thinking about it in terms of the fact that this is all blocks as opposed to only those in the main chain so there will always be more than whatever the current height is, but after thinking about it, `uint32` will provide more than enough space (2^31) for those extra blocks.